### ISSUE TITLE
F-Droid inclusion

### ISSUE THREAD

**Issue by FestplattenSchnitzel** [penn5/TrebleCheck#8]
[F-Droid](https://f-droid.org/) is an Android app store specifically for free/libre open-source apps. It would be great if your app could be released there, as it is the number one for getting FLOSS Android apps for many people.
F-Droid also builds all apps from source ([optionally even reproducible](https://f-droid.org/docs/Reproducible_Builds/)), so downloads from there can be trusted.

I can offer to care about the inclusion, since I already did a few of them. What I'll need is a "Yes, we allow you to do that!" and some changes in your repository (if you like I can even provide PRs).

If you want you can take a look at the [app developer FAQ](https://f-droid.org/docs/FAQ_-_App_Developers/) or the [quick start guide](https://f-droid.org/docs/FAQ_-_App_Developers/).

**Comment by penn5** [comment]
I did look before, their publishing procedures are prohibitive for me. If you can make it work, go ahead! Be careful, Play Billing may be a problem - you might need to introduce a new build variant in penn5/donations to disable it (or it might get removed by R8)

**Comment by penn5** [comment]
Maybe I wasn't clear. Yes, we allow you to do that!

**Comment by FestplattenSchnitzel** [comment]
It was clear but I didn't found the time yet. But thanks! :)

**Comment by FestplattenSchnitzel** [comment]
To get the app into F-Droid Data the following must be done:

- [ ] Add Fastlane structures to the repository as described [here](https://gitlab.com/-/snippets/1895688).
- [ ] Ensure literal values for versionCode and versionName at [app/build.gradle.kts#L61-L62](https://github.com/penn5/TrebleCheck/blob/dev/app/build.gradle.kts#L61-L62).
- [ ] Cover the above change(s) under a new or existing tag.

Would you be ok with adding a new build variant (=flavour) without the donations part? F-Droid displays donation links in-app, see

<img src="https://user-images.githubusercontent.com/45077355/127182022-90a3c738-9fe0-47e9-89a8-3129d5150f76.png" height="500">

**Comment by penn5** [comment]
> To get the app into F-Droid Data the following must be done:
> 
>     * [ ]  Add Fastlane structures to the repository as described [here](https://gitlab.com/-/snippets/1895688).
> 
>     * [ ]  Ensure literal values for versionCode and versionName at [app/build.gradle.kts#L61-L62](https://github.com/penn5/TrebleCheck/blob/dev/app/build.gradle.kts#L61-L62).
> 
>     * [ ]  Cover the above change(s) under a new or existing tag.
> 
> 
> Would you be ok with adding a new build variant (=flavour) without the donations part? F-Droid displays donation links in-app, see
> 
> <img alt="" height="500" src="https://user-images.githubusercontent.com/45077355/127182022-90a3c738-9fe0-47e9-89a8-3129d5150f76.png">

Yes that's all fine thanks. Note that some of the metadata for fastlane is auto-generated by penn5/poeditor-android and changes may be needed there. Is a constant version code a strict requirement? It would be handy to avoid this if possible.

**Comment by FestplattenSchnitzel** [comment]
> Is a constant version code a strict requirement? It would be handy to avoid this if possible.

Automatic metadata updates won't be possible, if there're no static values at `build.gradle`. Therefore the metadata would need to be updated manually for every new version.

To answer your question: It's not a strict requirement, but makes maintaining the app from F-Droid site a lot harder.

**Comment by penn5** [comment]
> > Is a constant version code a strict requirement? It would be handy to avoid this if possible.
> 
> Automatic metadata updates won't be possible, if there're no static values at `build.gradle`. Therefore the metadata would need to be updated manually for every new version.
> 
> To answer your question: It's not a strict requirement, but makes maintaining the app from F-Droid site a lot harder.

I don't mind adding fastlane metadata manually. I just would rather avoid having to deal with the gradle files manually too. I think that it should work fine if we leave it dynamic and just make sure to add correct fastlane metadata prior to tagging, right?

**Comment by FestplattenSchnitzel** [comment]
> I think that it should work fine if we leave it dynamic and just make sure to add correct fastlane metadata prior to tagging, right?

Would be OK. And, of course, we need a flavour without `Android Billing Client`; I think the donation part(s) of `TrebleCheck` can be dropped **in this flavour**, since F-Droid also displays donation links.
